src/librustc_metadata/creader.rs

```diff
@@ -855,6 +855,33 @@ impl<'a> CrateLoader<'a> {
         }
     }
 
+    fn inject_profiler_runtime(&mut self) {
+        if self.sess.opts.debugging_opts.profile {
+            let mut uses_std = false;
+            self.cstore.iter_crate_data(|_, data| {
+                if data.name == "std" {
+                    uses_std = true;
+                }
+            });
+
+            if uses_std {
```

Comment:
Mentioned above, we can remove this clause, right?

(just execute the code below if `-Z profile` is provided)